iis-wf/iis-wf-preprocessing/src/main/resources/eu/dnetlib/iis/wf/preprocessing/report/oozie_app/workflow.xml

```diff
@@ -37,6 +41,30 @@
             <arg>-Ipartial_reports=${input_partial_reports}</arg>
             <arg>-Oreport=${output_report}</arg>
         </java>
+        <ok to="report_name_generator" />
+        <error to="fail" />
+    </action>
+    
+    <action name="report_name_generator">
+        <java>
+            <main-class>eu.dnetlib.iis.common.report.ReportFileNameGenerator</main-class>
+            <arg>preprocessing</arg>
+            <capture-output />
+        </java>
+        <ok to="copy-report" />
+        <error to="fail" />
+    </action>
+    
+    <action name="copy-report">
+        <distcp xmlns="uri:oozie:distcp-action:0.1">
+            <job-tracker>${jobTracker}</job-tracker>
+            <name-node>${nameNode}</name-node>
+            <prepare>
+                <mkdir path="${nameNode}${reports_external_path}" />
+            </prepare>
+            <arg>${nameNode}${working_dir}/report.json</arg>
+            <arg>${nameNode}${reports_external_path}/${wf:actionData('report_name_generator')['report_filename']}</arg>
```

Comment:
> I considered using oozie timestamp() function.
> Documentation says that it will return string like: 1997-07-16T19:20:30.45Z. Which seemed too me quite long for a file name (seconds and fraction of seconds part).
> I don't know. If you think using timestamp() will be better then I will correct it.

`timestamp()` alone is too long but consider using mentioned `replaceAll()` function.

In plain java:
```
System.out.println("1997-07-16T19:20:30.45Z".replaceAll("(..)T(..):(..)\\:(\\d)+\\.(\\d)+Z$", "$1_$2-$3"));
```
gives:
```
1997-07-16_19-20
```
which is exactly what we need. 

Assuming `replaceAll()` workflow function works the same  - we are there, without involving `ReportFileNameGenerator`.